lib/gitter-adapter.js

```diff
@@ -306,6 +316,36 @@ Adapter.prototype.listUsers = function(channel, key) {
   });
 };
 
+Adapter.prototype.autoJoin = function() {
+  var gc = this.gitterClient;
+
+  gc.currentUser()
+  .then(function(user) {
+    return user.rooms();
+  })
+  .then(function(rooms) {
+    return rooms.map(function(room) {
+      return new GitterRoom(room, gc.client, gc.faye);
+    })
+    .filter(function(room) {
+      return !room.oneToOne;
+    })
+    .sort(function(a, b) { // sort descending (latest touched first)
+      return Date.parse(a.lastAccessTime) < Date.parse(b.lastAccessTime) ? 1 : -1;
```

Comment:
This does have the downside of not working with dates that are equal but this case probably won't come up and isn't critical.